### ISSUE TITLE
No initial state!

### ISSUE THREAD

**Issue by mehdi-cit** [davidkpiano/xstate#154]
Hello all,
I wonder if there's some best practice when the component you are working on does not have one obvious initial state.
Let me explain...
Let's say a component can have one of three different states. S1, S2 and S3.
However, we may "initiate" the component with either S1 or S2. Sometimes it needs to be initiated with S1 and other times with S2.
Whether we initiate with S1 or S2 depends on the state of the whole application (and not only on the local state of our component).
As I'm writing a react.js application, I thought of using the different life-cycle functions to set the initial state. So before rendering, I can read and inspect the application's global state and decide whether the component should be initiated (first rendered) with S1 or S2.
So basically, I introduce an initial state S0 that is never rendered. The component would transit to either S1 or S2 depending on the global state of the application.
Any thoughts? Maybe there's a better way?
Thanks in advance for all input!
Cheers~

**Comment by davidkpiano** [comment]
Hey @mehdi-cit - this is a very good question, and something that's been discussed before. The short answer is that you would have some default "preliminary" initial state, and then create an **eventless conditional transition** that transitions to the correct real "initial" state.

More info and discussion here: https://spectrum.chat/?t=c87d4b12-180d-46dc-bf48-c879ea3e731c

This made me realize that I need to add eventless transitions to the docs. You can also find more info (and the `xstate` implementation) here: https://statecharts.github.io/glossary/automatic-transition.html

**Comment by mehdi-cit** [comment]
Thanks David,
So, basically if a state SX has an eventless transition. Then this eventless transition is automatically "traversed" and our machine ends up in a different state.
We use condition to pick which eventless transition should be traversed.
I assume that's it and close the issue
